LivingLab.Core/DomainServices/EnergyUsageServices/EnergyUsageDomainService.cs

```diff
@@ -27,7 +27,7 @@ public EnergyUsageDomainService(ILabRepository labRepository, IEnergyUsageReposi
     /// <returns>EnergyUsageDTO</returns>
     public async Task<EnergyUsageDTO> GetEnergyUsage(EnergyUsageFilterDTO filter)
     {
-        // Grouping done here before SQLite doesn't support it
+        // Grouping done here because SQLite doesn't support it :(
         var logs = _energyUsageRepository
             .GetDeviceEnergyUsageByLabAndDate(filter.LabId, filter.Start, filter.End)
             .Result
```

Comment:
why not use await?
```cs
( await _energyUsageRepository
            .GetDeviceEnergyUsageByLabAndDate(filter.LabId, filter.Start, filter.End)
            ).GroupBy(log => log.LoggedDate.Date)
            .....
```
There is a performance difference as you scale to more and more simultaneous users, but I guess for this project it isn't very important.

<details>
  <summary>Detailed Explanation</summary>

- Accessing `Result` directly or using `.Wait()` causes the executing thread to block, leaving it occupied.
- `await` causes the function to suspend (like kotlin coroutines and suspend functions) and the thread is free to perform another task for another user before switching back and resuming the current function when the result is available.
</details>

Reply:
I didn't use await because I want to chain the query and aggregation into one line. Also, why I didn't do the grouping in the repo is because SQLite throws an error when I try to do this aggregation.

Reply:
you can chain the query and aggregation into one line even if you use await, if you wrap the await with brackets